tests/sentry/incidents/serializers/test_workflow_engine_action.py

```diff
@@ -1,51 +1,75 @@
     WorkflowEngineActionSerializer,
 )
 from sentry.incidents.models.alert_rule import AlertRuleTriggerAction
-from sentry.notifications.models.notificationaction import ActionTarget
 from sentry.testutils.cases import TestCase
 from sentry.testutils.helpers.datetime import freeze_time
-from sentry.workflow_engine.models import Action, ActionAlertRuleTriggerAction
-from sentry.workflow_engine.typings.notification_action import SentryAppIdentifier
+from sentry.workflow_engine.migration_helpers.alert_rule import (
+    migrate_alert_rule,
+    migrate_metric_action,
+    migrate_metric_data_conditions,
+)
 
 
 @freeze_time("2018-12-11 03:21:34")
 class TestActionSerializer(TestCase):
     def setUp(self) -> None:
         self.alert_rule = self.create_alert_rule()
-        self.trigger = self.create_alert_rule_trigger(alert_rule=self.alert_rule)
-        self.trigger_action = self.create_alert_rule_trigger_action(alert_rule_trigger=self.trigger)
-
-        self.action = self.create_action(
-            type=Action.Type.EMAIL.value,
-            config={
-                "target_type": ActionTarget.USER,
-                "target_identifier": str(self.user.id),
-            },
-        )
-        ActionAlertRuleTriggerAction.objects.create(
-            action_id=self.action.id,
-            alert_rule_trigger_action_id=self.trigger_action.id,
+        self.critical_trigger = self.create_alert_rule_trigger(
+            alert_rule=self.alert_rule, label="critical"
         )
-        self.data_condition_group = self.create_data_condition_group(organization=self.organization)
-        self.create_data_condition_group_action(
-            action=self.action, condition_group=self.data_condition_group
+        self.critical_trigger_action = self.create_alert_rule_trigger_action(
+            alert_rule_trigger=self.critical_trigger
         )
+        migrate_alert_rule(self.alert_rule)
+        migrate_metric_data_conditions(self.critical_trigger)
+        self.action, _, _ = migrate_metric_action(self.critical_trigger_action)
 
     def test_simple(self) -> None:
         serialized_action = serialize(self.action, self.user, WorkflowEngineActionSerializer())
+        assert serialized_action["id"] == str(self.critical_trigger_action.id)
+        assert serialized_action["alertRuleTriggerId"] == str(self.critical_trigger.id)
         assert serialized_action["type"] == "email"
         assert serialized_action["targetType"] == "user"
         assert serialized_action["targetIdentifier"] == str(self.user.id)
+        assert serialized_action["inputChannelId"] is None
+        assert serialized_action["integrationId"] is None
+        assert serialized_action["sentryAppId"] is None
+        assert serialized_action["dateCreated"] == self.action.date_added
+        assert serialized_action["desc"] == f"Send a notification to {self.user.email}"
+        assert serialized_action["priority"] == self.action.data.get("priority")
```

Comment:
nit: it might be nice to make the expected results a dictionary then have this be: `assert serialized_action == self.expected_result` -- then it will validate all the attributes at the same time as well -- which allows us to see any issues at the same time.